src/mds/Server.cc

```diff
@@ -526,7 +526,8 @@ void Server::handle_client_reclaim(const cref_t<MClientReclaim> &m)
     return;
   }
 
-  if (m->get_flags() & MClientReclaim::FLAG_FINISH) {
+  //let reclaim_session handle negative reclaim flags
+  if (((int32_t) m->get_flags()) > 0 && (m->get_flags() & MClientReclaim::FLAG_FINISH)) {
```

Comment:
```suggestion
  if (m->get_flags() & MClientReclaim::FLAG_FINISH) {
    if (m->get_flags() ^ MClientReclaim::FLAG_FINISH) {
      /* return EINVAL
    }
```

is what i'm suggesting.

Reply:
Okay, this can be useful but what happens in case we add new flags in future? Can it cause issues then?

Reply:
I can't predict the future but I don't think so. Even if we added a flag to be used with `FLAG_FINISH`, an older MDS should return `EINVAL` because it does not know what to do with it.

Reply:
Right, makes sense. I've pushed changes. PTAL.